### ISSUE TITLE
Passing custom Chrome profile directory (user data dir) not working on Mac and Linux

### ISSUE THREAD

**Issue by AJM-v** [cypress-io/cypress#18308]
### Current behavior

We would like to use a custom Chrome profile directory for our Cypress tests. The application we are testing uses a Web SQL database which needs to be loaded into the Chrome browser. We tried this command which should be a valid chromium argument.

`launchOptions.args.push('--user-data-dir=/Users/testuser/dev/chrome')`

But when we run Cypress we see that the Chrome Profile Path is being overwritten by Cypress:

![image](https://user-images.githubusercontent.com/91663159/135424857-e5ffd604-04ed-48c9-a4fd-35d22bead72c.png)

### Desired behavior

The Chrome Profile Path from the launchOptions `'--user-data-dir=/Users/testuser/dev/chrome'` is used.

### Test code to reproduce

Our setup:

- MacOs
- Chrome v93
- Cypress v8.4.0

The code we tried in our index.js (plugins folder):

```
module.exports = (on, config) => {
  
  on('before:browser:launch', (browser = {}, launchOptions) => {

    launchOptions.args.push('--user-data-dir=/Users/testuser/dev/chrome')

    launchOptions.args.push('--auto-open-devtools-for-tabs')
  
    return launchOptions
  })

};
```
Steps to reproduce:
1. Add launchOptions argument with --user-data-dir like mentioned above
2. Open Cypress dashboard (npx cypress open)
3. Run 1 test in Chrome
4. Chrome browser opens, type `chrome://version` in URL bar
5. --user-data-dir=/Users/testuser/dev/chrome is listed in the Command Line panel, but I also see --user-data-dir=/Users/xxxxx/Library/Application Support/Cypress/cy/production/browsers/chrome-stable/interactive listed, and that is the Profile Path which is being used by Cypress

### Cypress Version

8.4.0

### Other

Also tested on a Windows 10 environment and there this code is working fine when testing headed in Chrome.

**Comment by AlexMurphy** [comment]
@AJM-v Did you find a solution to this?

We have a use case that also requires custom Chrome profiles,

**Comment by AJM-v** [comment]
> @AJM-v Did you find a solution to this?
> 
> We have a use case that also requires custom Chrome profiles,

Nope, our dev team is creating a workaround for the app we are testing so that we don't have to use a custom chrome profile.

**Comment by lughino** [comment]
Same issue here on mac.
Cypress team could you have a look please?

**Comment by mike1936** [comment]
Same issue on Windows, both `--user-data-dir` and `--remote-debugging-port` settings are overwritten.

**Comment by lucassardois** [comment]
This issue still happen on cypress version 12 when using the run command from the CLI such as `npx cypress run --headed --no-exit --browser chrome`. I guess it's because when we run cypress it generate a new temp chrome profile to make the test isolated and this override this setting. 

Is there any workaround?

**Comment by zapplebee** [comment]
I can tell that Cypress is intentionally overwriting this arg here.

https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/browsers/chrome.ts#L692

But there seems to be no escape hatch to chose it yourself.

**Comment by PK-Tests** [comment]
Our case also requires this approach (we are setting up mic permissions before test).
I tried running this in `plugins/index.js`:
```js
export default (on, config) => {
    on('before:browser:launch', (browser = {}, launchOptions) => {
        launchOptions.args.push('--user-data-dir=../chromeProfile/Default');
    })
}
```

**Comment by kaviththiranga** [comment]
We have the same requirement. We are trying to run VSCode (Electron) with cypress and we need to override --user-data-dir

**Comment by drptbl** [comment]
This issue is a blocker for many things for us and it's still occurring.

**Comment by jennifer-shehane** [comment]
We did add an option recently to `IGNORE_CHROME_PREFERENCES`. You can see the code path here: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/browsers/chrome.ts#L111C19-L111C44

I haven't tried this specific use case, but it may be an option to opt out of the overwriting behavior that is causing problems with Cypress overwriting. Interested to hear if that works for anyone.

**Comment by localtechboy** [comment]
> We did add an option recently to `IGNORE_CHROME_PREFERENCES`. You can see the code path here: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/browsers/chrome.ts#L111C19-L111C44
> 
> I haven't tried this specific use case, but it may be an option to opt out of the overwriting behavior that is causing problems with Cypress overwriting. Interested to hear if that works for anyone.

I can confirm It doesn't work. Cypress still overwrites the `--user-data-dir` argument even with `IGNORE_CHROME_PREFERENCES` set to true.

**Comment by naridnevahgar** [comment]
Even with the IGNORE_CHROME_PREFERENCES set to true, I see that Chrome is launched with 2 --user-data-dir arguments. One with the value provided by me in the launchOptions and another is the temporary chrome profile created by Cypress. 


![chrome_version_output](https://github.com/user-attachments/assets/4f0e3fed-fd32-4c48-8575-8680e775fa01)

**Comment by sergiubologa** [comment]
> Even with the IGNORE_CHROME_PREFERENCES set to true, I see that Chrome is launched with 2 --user-data-dir arguments. One with the value provided by me in the launchOptions and another is the temporary chrome profile created by Cypress.
> 
> ![chrome_version_output](https://github.com/user-attachments/assets/4f0e3fed-fd32-4c48-8575-8680e775fa01)

same, it doesn't work for me either.

**Comment by ClownChu** [comment]
4 years and this is still happening?
Such a simple problem to get solved... if we specify a `--user-data-dir` (or any other option not required by Cypress) in the launch options, just don't replace it...

**Comment by amaify** [comment]
Is there any update on this? It's crucial for the project I'm working on.
